x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx

```diff
@@ -71,49 +75,63 @@ const AlertsTable: React.FunctionComponent<AlertsTableProps> = (props: AlertsTab
-              width: 75,
-              headerCellRender: () => {
-                return (
-                  <span data-test-subj="expandColumnHeaderLabel">
-                    {ALERTS_TABLE_CONTROL_COLUMNS_ACTIONS_LABEL}
-                  </span>
-                );
-              },
-              rowCellRender: (cveProps: EuiDataGridCellValueElementProps) => {
-                const { visibleRowIndex } = cveProps as EuiDataGridCellValueElementProps & {
-                  visibleRowIndex: number;
-                };
-                return (
-                  <EuiFlexGroup gutterSize="none" responsive={false}>
-                    <EuiFlexItem grow={false}>
-                      <EuiToolTip content={ALERTS_TABLE_CONTROL_COLUMNS_VIEW_DETAILS_LABEL}>
-                        <EuiButtonIcon
-                          size="s"
-                          iconType="expand"
-                          color="text"
-                          onClick={() => {
-                            setFlyoutAlertIndex(visibleRowIndex);
-                          }}
-                          data-test-subj={`expandColumnCellOpenFlyoutButton-${visibleRowIndex}`}
-                          aria-label={ALERTS_TABLE_CONTROL_COLUMNS_VIEW_DETAILS_LABEL}
-                        />
-                      </EuiToolTip>
-                    </EuiFlexItem>
-                  </EuiFlexGroup>
-                );
-              },
-            },
-          ]
-        : []),
+      {
+        id: 'expandColumn',
+        width: actionsColumnWidth,
+        headerCellRender: () => {
+          return (
+            <span data-test-subj="expandColumnHeaderLabel">
+              {ALERTS_TABLE_CONTROL_COLUMNS_ACTIONS_LABEL}
+            </span>
+          );
+        },
+        rowCellRender: (cveProps: EuiDataGridCellValueElementProps) => {
+          const { visibleRowIndex } = cveProps as EuiDataGridCellValueElementProps & {
+            visibleRowIndex: number;
+          };
+          const customActionsRow =
+            renderCustomActionsRow && renderCustomActionsRow(alerts[visibleRowIndex]);
+
+          return (
+            <EuiFlexGroup gutterSize="none" responsive={false}>
+              {props.showExpandToDetails && (
+                <EuiFlexItem grow={false}>
+                  <EuiToolTip content={ALERTS_TABLE_CONTROL_COLUMNS_VIEW_DETAILS_LABEL}>
+                    <EuiButtonIcon
+                      size="s"
+                      iconType="expand"
+                      color="primary"
+                      onClick={() => {
+                        setFlyoutAlertIndex(visibleRowIndex);
+                      }}
+                      data-test-subj={`expandColumnCellOpenFlyoutButton-${visibleRowIndex}`}
+                      aria-label={ALERTS_TABLE_CONTROL_COLUMNS_VIEW_DETAILS_LABEL}
+                    />
+                  </EuiToolTip>
+                </EuiFlexItem>
+              )}
+              {customActionsRow && customActionsRow}
```

Comment:
nit -> That's just look weird to my eyes, but it works
```suggestion
              {renderCustomActionsRow && renderCustomActionsRow(alerts[visibleRowIndex])}
```

Reply:
I merged it manually, thanks :)